test/functional/ui/popupmenu_spec.lua

```diff
@@ -5935,6 +5935,114 @@ describe('builtin popupmenu', function()
         ]])
         feed('<C-E><Esc>')
       end)
+
+      -- oldtest: Test_pum_complete_with_special_characters()
+      it('multi-line completion', function()
+        exec([[
+          func Omni_test(findstart, base)
+            if a:findstart
+              return col(".")
+            endif
+            return [#{word: "func ()\n\t\nend", abbr: "function ()",}, #{word: "foobar"}, #{word: "你好\n\t\n我好"}]
+          endfunc
+          set omnifunc=Omni_test
+        ]])
+
+        feed('S<C-X><C-O>')
+        screen:expect([[
+          func ()                         |
+                                          |
+          end^                             |
+          {s:function ()    }{1:                 }|
+          {n:foobar         }{1:                 }|
+          {n:你好^@  ^@我好 }{1:                 }|
```

Comment:
Why is it rendering as `^@` instead of linebreak? I thought the entire purpose of https://github.com/vim/vim/issues/2505 was to insert linebreaks.

Reply:
What is "rendering as linebreak"?

Reply:
Oh I see these tests are checking the menu representation, not the inserted text. I was expecting a test that verifies the inserted text has multiple lines instead of one line with `^@` chars.

Reply:
Isn't it right here?